Tests/Web5Tests/Dids/DIDDHTTests.swift

```diff
@@ -0,0 +1,191 @@
+import Mocker
+import XCTest
+
+@testable import Web5
+
+final class DIDDHTTests: XCTestCase {
+
+    func test_resolve_publishedDID_withSingleVerificationMethod() async throws {
+        Mock(
+            url: URL(string: "https://diddht.tbddev.org/pjiiw7ibn6t9k1mkknkowjketa8chksgwzkt5uk8798epux1386o")!,
```

Comment:
if these tests rely on DIDs being present on this node it will work for some time...then break when we move to our prod node.

separately I believe it's crucial to implement tests for the two vectors the spec currently has https://did-dht.com/#test-vectors

Reply:
I copied these tests from the `web5-js` repo [here](https://github.com/TBD54566975/web5-js/blob/main/packages/dids/tests/methods/did-dht.spec.ts#L630-L769).

In terms of the test-vectors defined in the spec, I'm a little confused as to what exactly they are representing. It _looks_ like they are _not_ resolution vectors, but instead are importing keys, and verifying a particular DIDDocument is constructed from them. Creation/Importing will be coming in a later PR, and should be able to handle those. Do let me know if I'm misunderstanding those though!

Reply:
they are not exactly resolution vectors, you are correct. they make sure you can construct a consistent document and packet given a set of inputs.

I mention it now because you [have implemented fromDNSPacket](https://github.com/TBD54566975/web5-swift/pull/15/files#diff-b6ab4a7a98ae19ce0e1b7bb74c0169cb09a49a24c03e736b2454ff2254ee52a7R228) which could exercise the listed vectors DNS packets and compare them to the resulting DID document.

Reply:
Added these vectors in the following commit: [93351ac](https://github.com/TBD54566975/web5-swift/pull/15/commits/93351acdf843e6b683f285ea5c5bd61042a059b9)

These vectors are kind of a deviation from the existing `resolve` vectors in the web5-spec [here](https://github.com/TBD54566975/web5-spec/blob/main/test-vectors/did_dht/resolve.json), so I decided to just implement them directly in Swift for the time being. Eventually, I think these should work their way into the `web5-spec` repo though, so that we ensure they pass across all SDKs. Just a matter of aligning on input/output formats!

Reply:
thank you @amika-sq I noted [a comment here](https://github.com/TBD54566975/web5-spec/issues/11#issuecomment-1948958477) and will make sure to get these updated.